### PR TITLE
Documentation for ancestry functionality

### PR DESCRIPTION
**Pull request by kevgo**
@charlierudolph @allewun @ricmatsui 

Updates the documentation to include the new nested feature branch functionality.

### CODE REVIEW

File: documentation/commands/git-new-pull-request.md [hunk 1]
```diff
@@ -12,5 +12,5 @@ git new-pull-request
 
 #### DESCRIPTION
 
-Opens a browser to a new pull request for the current branch.
+Opens a browser window to the "new pull request" page for the current branch on the code hosting platform.
```
Comment (charlierudolph) [inline documentation/commands/git-new-pull-request.md]: What you do think about 
`to the form for a new pull request` 
instead of 
`to the "new pull request" page`

File: documentation/commands/git-sync.md [hunk 2]
```diff
@@ -15,9 +15,9 @@ git sync (--abort | --continue | --skip)
 Synchronizes the current branch with the rest of the world.
 
 When run on a feature branch
-* syncs the main branch
+* syncs all ancestor branches including the main branch
```
Comment (charlierudolph) [inline documentation/commands/git-sync.md]: do we need to mention `including the main branch`? Since the main branch is an ancestor of all feature branches I think thats implied

File: man/man1/git-hack.1 [hunk 3]
```diff
@@ -1,7 +1,7 @@
 .TH "GIT-HACK" "1" "04/02/2015" "Git Town 0\&.6\&.0" "Git Town Manual"
 
 .SH "NAME"
-git hack \- cut a new feature branch off the main branch
+git hack \- create a new feature branch
 
 .SH "SYNOPSIS"
 \fIgit hack\fR <branchname> [parent branch name]
```
Comment (charlierudolph) [inline man/man1/git-hack.1]: unrelated but I think this should be updated from `[parent branch name]` to `[<parent_branch_name>]`

File: man/man1/git-hack.1 [hunk 4]
```diff
@@ -9,9 +9,11 @@ git hack \- cut a new feature branch off the main branch
 \fIgit hack\fR (--abort | --continue)
 
 .SH "DESCRIPTION"
-Sync the main branch and create a new feature branch with the given name.
+Syncs the main branch,
```
Comment (charlierudolph) [inline man/man1/git-hack.1]: Doesn't this now sync the parent branch?

File: documentation/commands/git-ship.md [hunk 5]
```diff
@@ -13,15 +13,19 @@ git ship (--abort | --continue)
 
 #### DESCRIPTION
 
-Squash merges the current branch, or `<branchname>` if given, into the main branch, leading to linear history on the main branch.
+Squash merges the current branch, or `<branchname>` if given,
```
Comment (allewun) [inline documentation/commands/git-ship.md]: hyphenate squash-merge

File: documentation/commands/git-new-pull-request.md (hunk 1 above)
Comment (charlierudolph) [inline documentation/commands/git-new-pull-request.md]: At the moment this is unrelated to this PR. Should this be updated to mention parent branch somewhere?

Comment (kevgo) [inline documentation/commands/git-new-pull-request.md]: I like the new phrasing!

I have added a sentence that explains that the PR only shows changes against the immediate parent branch. What do you think about this?

File: documentation/commands/git-ship.md (hunk 5 above)
Comment (kevgo) [inline documentation/commands/git-ship.md]: Hyphenated everywhere

File: documentation/commands/git-sync.md (hunk 2 above)
Comment (kevgo) [inline documentation/commands/git-sync.md]: True. Removed

File: man/man1/git-hack.1 (hunk 3 above)
Comment (kevgo) [inline man/man1/git-hack.1]: I'm following http://docopt.org, where in my understanding square brackets denote optional arguments, and angled brackets required arguments.

File: man/man1/git-hack.1 (hunk 4 above)
Comment (kevgo) [inline man/man1/git-hack.1]: Right. Fixed

**Comment by kevgo** [comment]
all comments addressed

File: man/man1/git-hack.1 (hunk 3 above)
Comment (charlierudolph) [inline man/man1/git-hack.1]: I believe angled brackets only imply user input. I believe if we fed this into docopt the `parent branch name` cannot be any string but would need to be exactly `parent branch name`.

Comment (kevgo) [inline man/man1/git-hack.1]: Ah, that makes sense. Fixed

**Comment by charlierudolph** [comment]
Looks like this PR should be closed and a new PR needs to opened up against master

**Comment by kevgo** [comment]
Since this is only documentation, and we didn't have much other comments in the last 2 weeks, I shipped it. We can always update documentation easily.
